src/modules/wifi/scan_hosts.cpp

```diff
@@ -5,140 +5,100 @@
 #include "core/wifi_common.h"
 #include "scan_hosts.h"
 #include "clients.h"
+#include <ESPping.h>
+#include <HTTPClient.h>
+#include <WiFi.h>
 
 //thx to 7h30th3r0n3, which made scanHosts faster using ARP
 
-std::vector<IPAddress> hostslist;
+static std::vector<Host> hostslist;
+static uint32_t localIp;
 
-void read_arp_table(char * from_ip, int read_from, int read_to, std::vector<IPAddress>& hostslist) {
-  Serial.printf("Reading ARP table from: %d to %d\n", read_from, read_to);
-  for (int i = read_from; i <= read_to; i++) {
-    char test[32];
-    sprintf(test, "%s%d", from_ip, i);
-    ip4_addr_t test_ip;
-    ipaddr_aton(test, (ip_addr_t*)&test_ip);
+// TODO: resolve clients name when in host mode through dhcp
 
-    const ip4_addr_t *ipaddr_ret = NULL;
-    struct eth_addr *eth_ret = NULL;
-    if (etharp_find_addr(NULL, &test_ip, &eth_ret, &ipaddr_ret) >= 0) {
-      IPAddress foundIP;
-      foundIP.fromString(ipaddr_ntoa((ip_addr_t*)&test_ip));
-      hostslist.push_back(foundIP);
-      //tft.println(foundIP.toString());
-      Serial.printf("Adding found IP: %s\n", ipaddr_ntoa((ip_addr_t*)&test_ip));
+// TODO: move to a config
+// gives hosts time to respond
+// from the tests with the value under 4k high-end hosts in subnet \24 are not able to respond
+TickType_t beforeReadingArpDelay = 5000u / portTICK_PERIOD_MS; 
```

Comment:
I tried to make less delay, maybe this can be smaller.
The current release build can probe the 254 hosts in approximately 4 seconds, after your PR even before removing the double `send_arp`, it looks like it takes approximately 10 seconds to probe all hosts...
Im not sure if thats the cause of it.

Reply:
Yeah, it was combination of 4s delay between sending and reading + 40ms between each
I've completely reworked reading function to consider cached requests limitation, so now it's much quicker

If it's not a problem, could you check on your network that all required hosts are now found? Because my initial reasoning behind adding that timeout is that not all hosts were found